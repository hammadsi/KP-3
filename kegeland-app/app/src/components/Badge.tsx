import React from 'react';
import {Badge as PaperBadge, useTheme} from 'react-native-paper';

export type BadgeProps = React.ComponentProps<typeof PaperBadge>;

/**
 * Component for rendering a badge.
 * @param props the props
 * @see {@link BadgeProps}
 * @see {@link Badge}
 */
const Badge: React.FC<BadgeProps> = (props) => {
  const {colors} = useTheme();
  return (
    <PaperBadge
      {...props}
      style={[
        {backgroundColor: colors.primary, color: colors.text},
        props.style,
      ]}
    />
  );
};

export default Badge;
