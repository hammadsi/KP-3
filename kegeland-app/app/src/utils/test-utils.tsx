/* istanbul ignore file */
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {render, RenderOptions} from '@testing-library/react-native';
import {Provider as ReduxProvider} from 'react-redux';

import {store} from '~state/store';
import {lightTheme} from '~constants/theme';
import {initialStore, mockStore} from '~state/ducks/__mocks__/store.mock';

type TestWrapperProps = {
  children: React.ReactNode;
  store?: typeof store;
};

/**
 * Wrapper component with redux state. Used when testing components
 * which requires redux.
 * @param param0 test wrapper props @see {@link TestWrapperProps}
 * @returns
 */
export const TestWrapper: React.FC<TestWrapperProps> = ({
  children,
  ...props
}) => {
  return (
    <ReduxProvider store={props.store || mockStore(initialStore)}>
      <PaperProvider theme={lightTheme}>
        <NavigationContainer theme={lightTheme}>{children}</NavigationContainer>
      </PaperProvider>
    </ReduxProvider>
  );
};

/**
 * Custom test renderer which wraps components with a redux store provider
 * @param ui the component to render
 * @param options renderer options
 * @returns test rendered componnet
 */
export const customRender = <T extends any>(
  ui: React.ReactElement<T>,
  options?: {
    wrapperProps?: {
      store: any;
    };
    renderOptions?: RenderOptions;
  },
) =>
  render(ui, {
    wrapper: (props) => <TestWrapper {...props} {...options?.wrapperProps} />,
    ...options?.renderOptions,
  });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
