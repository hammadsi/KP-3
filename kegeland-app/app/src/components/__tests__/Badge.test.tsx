import React from 'react';

import {render} from '~utils/test-utils';

import Badge from '../Badge';

describe('Test Badge-component', () => {
  it('should render correctly', () => {
    const component = <Badge>1</Badge>;
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
