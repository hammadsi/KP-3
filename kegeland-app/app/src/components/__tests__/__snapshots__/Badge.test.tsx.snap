// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Badge-component should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    collapsable={false}
    nativeID="animatedComponent"
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#00b0ff",
        "borderRadius": 10,
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 10,
        "fontWeight": "400",
        "height": 20,
        "lineHeight": 10,
        "minWidth": 20,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 4,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    1
  </Text>
</View>
`;
